src/lib.rs

```diff
@@ -39,6 +46,95 @@ impl UploadMetadata {
     }
 }
 
+#[derive(Parser, Debug)]
+#[clap(name = "merge app")]
+pub struct Opt {
+    #[clap(long)]
+    pub merger_config_path: PathBuf,
+
+    #[clap(long)]
+    pub external_config_path: Option<PathBuf>,
+
+    #[clap(long)]
+    pub relative_search_path: Option<PathBuf>,
+
+    #[clap(long)]
+    pub config_path: PathBuf,
+
+    #[clap(long)]
+    pub manifest_path: PathBuf,
+
+    #[clap(long)]
+    pub manifest_sha256_path: PathBuf,
+
+    #[clap(long)]
+    pub upload_metadata_path: PathBuf,
+}
+
+pub async fn merge_main(opt: Opt) -> Result<(), anyhow::Error> {
+    let pusher_config = MergeConfig::parse_file(&opt.merger_config_path)?;
+
+    let external_execution_config = match &opt.external_config_path {
```

Comment:
we can't use `.map` here or something?

ahh maybe not because of the `?` man... `?` is really a very poor man's traverse.